akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -682,6 +683,18 @@ public void apply(String elem) {
     probe.expectMsgEquals("C");
   }
 
+  @Test
+  public void mustBeAbleToUseCollectType() throws Exception{
+    final TestKit probe = new TestKit(system);
+    final Iterable<FlowSpec.Apple> input = Collections.singletonList(new FlowSpec.Apple());
```

Comment:
It would be good if it actually contained other things than apples so we would know that the collect actually works. The current test data would make the test pass also with `Flow(t -> t)`.

Reply:
Okay

Reply:
I agree, `Orange extends Fruit`, `Apple extends Fruit`